### PR TITLE
AC-843 Added possible NPE checks at PatientComparator

### PR DESCRIPTION
**Pull request by LuGO0**
## Description of what I changed
1. Added required NPE checks for the comparator.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-843

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/PatientComparator.kt [hunk 1]
```diff
@@ -89,14 +87,16 @@ class PatientComparator {
...
         var score = 0
-        if (Objects.equal(newPatient.name.givenName, existingPatient.name.givenName)) {
-            score += 1
-        }
-        if (Objects.equal(newPatient.name.familyName, existingPatient.name.familyName)) {
-            score += 1
-        }
-        if (Objects.equal(newPatient.name.middleName, existingPatient.name.middleName)) {
-            score += 1
+        if (existingPatient.name != null && newPatient.name != null) {
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/PatientComparator.kt:90]: did you tried running these changes??? And by any chance, were you able to replicate the error as stated bt ssmoke ???

Comment (LuGO0) [reply openmrs-client/src/main/java/org/openmrs/mobile/utilities/PatientComparator.kt:90]: no @rishabh-997 I wasnt able to replicate the issue as raised by ssmusoke, but having NPE checks wont hurt so I added .

**Review by rishabh-997 — APPROVED** [review]
LGTM...

**Review by f4ww4z — APPROVED** [review]
@LuGO0 LGTM

**Pull request merged**